read-fonts/src/tables/postscript/dict.rs

```diff
@@ -0,0 +1,718 @@
+        CidCount => Entry::CidCount(stack.pop_i32()? as u32),
+        UidBase => Entry::UidBase(stack.pop_i32()?),
+        FontName => Entry::FontName(stack.pop_i32()?.into()),
+        StdHw => Entry::StdHw(stack.pop_fixed()?),
+        StdVw => Entry::StdVw(stack.pop_fixed()?),
+        DefaultWidthX => Entry::DefaultWidthX(stack.pop_fixed()?),
+        NominalWidthX => Entry::NominalWidthX(stack.pop_fixed()?),
+        StemSnapH => {
+            stack.apply_delta_prefix_sum();
+            Entry::StemSnapH(StemSnaps::new(stack.fixed_values()))
+        }
+        StemSnapV => {
+            stack.apply_delta_prefix_sum();
+            Entry::StemSnapV(StemSnaps::new(stack.fixed_values()))
+        }
+        ForceBold => Entry::ForceBold(stack.pop_i32()? != 0),
+        InitialRandomSeed => Entry::InitialRandomSeed(stack.pop_i32()?),
+        // Blend is handled at the layer above
+        Blend => unreachable!(),
+    })
+}
+
+/// <https://gitlab.freedesktop.org/freetype/freetype/-/blob/80a507a6b8e3d2906ad2c8ba69329bd2fb2a85ef/src/psaux/psblues.h#L141>
+const MAX_BLUE_VALUES: usize = 7;
+
+/// Operand for the `BlueValues`, `OtherBlues`, `FamilyBlues` and
+/// `FamilyOtherBlues` operators.
+///
+/// These are used to generate zones when applying hints.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+pub struct Blues {
+    values: [(Fixed, Fixed); MAX_BLUE_VALUES],
+    len: u32,
+}
+
+impl Blues {
+    fn new(values: impl Iterator<Item = Fixed>) -> Self {
+        let mut blues = Self::default();
+        let mut stash = Fixed::ZERO;
+        for (i, value) in values.take(MAX_BLUE_VALUES * 2).enumerate() {
+            if (i & 1) == 0 {
+                stash = value;
+            } else {
+                blues.values[i / 2] = (stash, value);
+                blues.len += 1;
+            }
+        }
+        blues
+    }
+
+    pub fn values(&self) -> &[(Fixed, Fixed)] {
+        &self.values[..self.len as usize]
+    }
+}
+
+/// Summary: older PostScript interpreters accept two values, but newer ones
+/// accept 12. We'll assume that as maximum.
+/// <https://adobe-type-tools.github.io/font-tech-notes/pdfs/5049.StemSnap.pdf>
+const MAX_STEM_SNAPS: usize = 12;
+
+/// Operand for the `StemSnapH` and `StemSnapV` operators.
+///
+/// These are used for stem darkening when applying hints.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+pub struct StemSnaps {
+    values: [Fixed; MAX_STEM_SNAPS],
+    len: u32,
+}
+
+impl StemSnaps {
+    fn new(values: impl Iterator<Item = Fixed>) -> Self {
+        let mut snaps = Self::default();
+        for (value, target_value) in values.take(MAX_STEM_SNAPS).zip(&mut snaps.values) {
+            *target_value = value;
+            snaps.len += 1;
+        }
+        snaps
+    }
+
+    pub fn values(&self) -> &[Fixed] {
+        &self.values[..self.len as usize]
+    }
+}
+
+fn parse_int(cursor: &mut Cursor, b0: i32) -> Result<i32, Error> {
```

Comment:
why do we pass in `b0` as an i32?

Reply:
It’s used as i32 in some of the match arms. This is probably an artifact of my extracting the code into a function. I’ll switch to u8 and do the cast internally.